### DISCUSSION TITLE
Documenting multiple return values

### DISCUSSION THREAD

**Discussion by corykinney** [mkdocstrings/mkdocstrings#300] [Q&A] [answered]
I'm having trouble documenting multiple return values, and I was hoping someone could help. Let's just say I have the following file `foo.py`:

```python
def foo() -> tuple[float, float]:
    """Example function.
    
    Returns:
        a: first return value
        b: second return value
    """

    ...

    return a, b
```

When I run mkdocs using the autodoc syntax `::: foo.foo` I get the returns table as a single row with the `tuple` type and the description contains both of my intended variable descriptions. Am I making a mistake with the docstring formatting/syntax or is it not possible to unpack these multiple returns into different rows?

EDIT: Reference issue #301

**Comment by pawamoy** [comment, marked as answer]
It is indeed not possible to unpack the return value of the docstrings into a table, as it is impossible (as in the current situation) to predict what the return value is composed of. But you can still use Markdown to improve the formatting, for example by using a list:

```python
def foo() -> tuple[float, float]:
    """Example function.
    
    Returns:
        - `a`: first return value.
        - `b`: second return value.
    """

    ...

    return a, b
```

**Reply by corykinney** [reply]
The Markdown formatting you suggested definitely improves the readability of the result. However, I fail to see why it shouldn't be possible with type hinting. If I have the following function

```python
def foo(a: float, b: float) -> tuple[float, float]:
    """Example function.

    Arguments:
        a: first argument description.
        b: second argument description.

    Returns:
        a: first return value description.
        b: second return value description.
    """
    return a, b
```

then mkdocstrings knows to use the type hint from the function signature and create a table with two rows with the matching variable name, type, and description. If the function signature also includes type hinting of the return value, in this case `tuple[float, float]`, then I don't see why it shouldn't be possible to make a table for the return values as well. I understand if this is just not implemented yet, but I think it would be a useful feature that produces clearer documentation.

**Reply by pawamoy** [reply]
I didn't think of using the return annotation, thank you for expanding your request. Then yeah, multiple return values could be supported when the return annotation is a tuple.

**Reply by corykinney** [reply]
@pawamoy should I open a feature request in the issue tracker?

**Reply by pawamoy** [reply]
Yes please!
